cloud/deployment/deployment.go

```diff
@@ -274,105 +291,275 @@ func Create(label, workspaceID, description, clusterID, runtimeVersion, dagDeplo
+		if !resourcesValid {
+			return ErrInvalidResourceRequest
+		}
+		var requestedExecutor astroplatformcore.CreateHybridDeploymentRequestExecutor
+		if executor == CeleryExecutor {
+			requestedExecutor = astroplatformcore.CreateHybridDeploymentRequestExecutorCELERY
+		}
+		if executor == KubeExecutor {
+			requestedExecutor = astroplatformcore.CreateHybridDeploymentRequestExecutorKUBERNETES
+		}
+		hybridDeploymentRequest := astroplatformcore.CreateHybridDeploymentRequest{
+			AstroRuntimeVersion: runtimeVersion,
+			Description:         &description,
+			Name:                name,
+			Executor:            requestedExecutor,
+			IsCicdEnforced:      isCicdEnforced,
+			IsDagDeployEnabled:  dagDeployEnabled,
+			ClusterId:           clusterID,
+			WorkspaceId:         workspaceID,
+
+			Scheduler: astroplatformcore.DeploymentInstanceSpecRequest{
+				Au:       schedulerAU,
+				Replicas: schedulerReplicas,
+			},
+			Type: astroplatformcore.CreateHybridDeploymentRequestTypeHYBRID,
+		}
+		if executor == CeleryExecutor {
+			hybridDeploymentRequest.WorkerQueues = &defautWorkerQueue
+		} else {
+			hybridDeploymentRequest.TaskPodNodePoolId = &nodePools[0].Id
+		}
+		err = createDeploymentRequest.FromCreateHybridDeploymentRequest(hybridDeploymentRequest)
+		if err != nil {
+			return err
 		}
 	}
 
-	// Create request
-	d, err := client.CreateDeployment(createInput)
+	d, err := CoreCreateDeployment(organizationID, createDeploymentRequest, corePlatformClient)
 	if err != nil {
 		return err
 	}
 
 	if waitForStatus {
-		err = HealthPoll(d.ID, workspaceID, sleepTime, tickNum, timeoutNum, coreClient)
+		err = HealthPoll(d.Id, workspaceID, sleepTime, tickNum, timeoutNum, corePlatformClient)
 		if err != nil {
-			errOutput := createOutput(workspaceID, clusterType, &d)
+			errOutput := createOutput(workspaceID, &d)
 			if errOutput != nil {
 				return errOutput
 			}
 			return err
 		}
 	}
 
-	err = createOutput(workspaceID, clusterType, &d)
+	err = createOutput(workspaceID, &d)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func createOutput(workspaceID, clusterType string, d *astro.Deployment) error {
+func createOutput(workspaceID string, d *astroplatformcore.Deployment) error {
 	tab := newTableOut()
 
-	runtimeVersionText := d.RuntimeRelease.Version + " (based on Airflow " + d.RuntimeRelease.AirflowVersion + ")"
-	clusterName := d.Cluster.Name
-	releaseName := d.ReleaseName
-	if organization.IsOrgHosted() {
-		if clusterType == standard {
-			clusterName = d.Cluster.Region
-		}
-		releaseName = notApplicable
-	}
-	tab.AddRow([]string{d.Label, releaseName, clusterName, d.ID, runtimeVersionText, strconv.FormatBool(d.DagDeployEnabled), strconv.FormatBool(d.APIKeyOnlyDeployments)}, false)
-	deploymentURL, err := GetDeploymentURL(d.ID, workspaceID)
+	runtimeVersionText := d.RuntimeVersion + " (based on Airflow " + d.AirflowVersion + ")"
+	clusterName := notApplicable
```

Comment:
wouldn't it be applicable for dedicated and hybrid clusters
seems like as of right now, clusterName will always be `NA`

Reply:
fixed it